### PR TITLE
Add CloudFormation template for managing plugin IAM users

### PR DESCRIPTION
**Pull request by camerski**
*Description of changes:*
Add a CloudFormation template for managing the IAM users we allow to upload plugin artifacts to the staging bucket.

*Test Results:*
Manually tested in a burner account. Verified that users are created correctly. Used the IAM policy simulator to verify that they can put to the right locations and cannot get or delete objects, or create or delete buckets.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: release-tools/cloudformation/plugin-iam-users.yml [hunk 1]
```diff
@@ -0,0 +1,177 @@
+Description: Template for setting up IAM users for each plugin repository so they can publish artifacts into the staging bucket
+Parameters:
+  BucketName:
+    Type: String
+    Description: The name of the staging bucket
+    Default: staging.artifacts.opendistroforelasticsearch.amazon.com
```
Comment (gaiksaya) [inline release-tools/cloudformation/plugin-iam-users.yml:6]: Safe to name the bucket?

File: release-tools/cloudformation/plugin-iam-users.yml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      Policies:
+        - PolicyName: S3_Write_Access
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              Effect: Allow
+              Action: "s3:PutObject"
+              Resource:
+                - !Sub "arn:aws:s3:::${BucketName}/snapshot/elasticsearch-plugins/sql/*"
+                - !Sub "arn:aws:s3:::${BucketName}/snapshot/clients/sqlcli/*"
```
Comment (gaiksaya) [inline release-tools/cloudformation/plugin-iam-users.yml:47]: would also need to `kibana-plugins/sql-workbench/*` access

File: release-tools/cloudformation/plugin-iam-users.yml (hunk 1 above)
Comment (camerski) [reply release-tools/cloudformation/plugin-iam-users.yml:6]: Yeah, I think so. It's not a secret.

File: release-tools/cloudformation/plugin-iam-users.yml (hunk 2 above)
Comment (camerski) [reply release-tools/cloudformation/plugin-iam-users.yml:47]: Added

**Review by gaiksaya — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**